### PR TITLE
Add symfony 3.0 support

### LINKED ISSUE #115: FlattenException deprecated
Using deprecated class Symfony\Component\HttpKernel\Exception\FlattenException | 572  | Deprecated in 2.3, to be removed in 3.0. Use the same class from the Debug component instead. |

I think we should replace with this one
use Symfony\Component\Debug\Exception\FlattenException;

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR adds symfony 3.0 support and removes the usage of the deprecated ``FlattenException`` class.

Fix #115, #121, #116, #73

### CODE REVIEW

**Comment by Ener-Getick** [comment]
The build fail is not related to this PR.
(see https://travis-ci.org/schmittjoh/JMSJobQueueBundle/jobs/97841916)

**Comment by temp** [comment]
I've got the same error with travis in my PR #70.

File: composer.json [hunk 1]
```diff
@@ -12,37 +12,36 @@
         }
     ],
     "require": {
-        "php": "^5.5.0",
-        "symfony/framework-bundle": ">=2.1,<3.0-dev",
-        "doctrine/common": ">=2.3,<3.0",
-        "jms/di-extra-bundle": ">=1.1,<2.0"
+        "php": "^5.5|^7.0",
+        "symfony/framework-bundle": "^2.3|^3.0",
```
Comment (schmittjoh) [inline composer.json]: 2.1 should still work afaics, no? any reason why we need 2.3 here?

Comment (Ener-Getick) [inline composer.json]: Symfony 2.1 and 2.2 are not supported anymore so IMO we should not continue support them too.

Comment (schmittjoh) [inline composer.json]: If we don't need the higher version, let's keep the old ``2.1`` constraint.

Comment (Ener-Getick) [inline composer.json]: Changed

**Comment by Ener-Getick** [comment]
@temp the tests error message really need to be improved... :/

**Comment by whyte624** [comment]
Waiting for this pull request to be pulled. :+1:

**Comment by Ener-Getick** [comment]
@whyte624 unfortunately I don't know how to fix the tests :-/

**Comment by whyte624** [comment]
@Ener-Getick right now I'm using your SF3 branch.
So could you please fix `kernel.root_dir` in commands?
For example, jms/job-queue-bundle/Command/RunCommand.php:433
change `$this->getContainer()->getParameter('kernel.root_dir')` to something which returns bin directory.

**Comment by Ener-Getick** [comment]
@whyte624 the ``RunCommand`` now uses a ``Finder`` to find the ``console`` command ;-)

**Comment by e-doceo** [comment]
+1

**Comment by khasinski** [comment]
:+1:

**Comment by whyte624** [comment]
Is it possible to make it compatible with php 7?

**Comment by ruslan-polutsygan** [comment]
any news when this can be merged?

File: Command/RunCommand.php [hunk 2]
```diff
@@ -428,9 +429,25 @@ private function getCommandProcessBuilder()
             $pb->add('exec');
         }
 
+        // Localize the `console` command
```
Comment (schmittjoh) [inline Command/RunCommand.php]: Can we detect the console command based on how this command itself was started?

**Comment by ruslan-polutsygan** [comment]
@Ener-Getick 
can you merge master in your branch, or make rebase? 
this is fixed in master but still an issue in your branch https://github.com/schmittjoh/JMSJobQueueBundle/issues/123

File: Command/RunCommand.php (hunk 2 above)
Comment (Ener-Getick) [inline Command/RunCommand.php]: This doesn't look safe if the code is executed in a test case or from a web app.

Comment (Ener-Getick) [inline Command/RunCommand.php]: How would you do that?

**Comment by Ener-Getick** [comment]
done @ruslan-polutsygan

File: Command/RunCommand.php (hunk 2 above)
Comment (schmittjoh) [inline Command/RunCommand.php]: If you check the content of ``$_SERVER``, there should be everything you need.

Comment (Ener-Getick) [inline Command/RunCommand.php]: This looks much more complicated as the user can use either ``php console`` or ``console`` so we would have to parse the input.

**Comment by jiri-jagos** [comment]
May I ask when approximately is this going to be merged into the master please?

**Comment by dextervip** [comment]
+1

**Comment by stephanvierkant** [comment]
Any news on this? Where are we waiting for?

**Comment by dextervip** [comment]
It looks like repo owner is not reviewing issues and pull requests anymore.

**Comment by Ener-Getick** [comment]
@stephanvierkant @dextervip you can still try to send him a message on twitter, he may have too many notifications ;-)

**Comment by dextervip** [comment]
@Ener-Getick Are you using this pull request in your production env with sf3?

**Comment by Ener-Getick** [comment]
@dextervip no i don't, but it seems that others do (my repo has been forked several times and some issues have been reported).

**Comment by dextervip** [comment]
There is one test failing in travis build, Any ideas? I guess It should be fixed before we hit up

**Comment by Ener-Getick** [comment]
@dextervip in fact this is already failing in the current state and the test pass on my machine so i have no idea from what it comes.

**Comment by dextervip** [comment]
@Ener-Getick Yes, just noticed it.

Maybe we should ask to be created a 1.x branch to maintain legacy bundle support and use master to develop new sf3 support, new BC features and release 2.x versions. What do you think guys?

**Comment by Ener-Getick** [comment]
@dextervip this PR is fully bc so i don't think we need to release a new major version.

File: Command/RunCommand.php (hunk 2 above)
Comment (nclavaud) [inline Command/RunCommand.php]:
```php
realpath($_SERVER['SCRIPT_FILENAME'])
// or
realpath($_SERVER['argv'][0])
```
should do the trick, whether the `php` executable is mentioned or not.

This looks safer than expecting to find a single file called `console` somewhere under the root dir - which is a convention that not all projects might follow.

Comment (Ener-Getick) [inline Command/RunCommand.php]: @nclavaud could you send a PR on my repo please ?

**Comment by nullziu** [comment]
Eagerly waiting for this to get merged! :+1:

**Comment by dmkuznetsov** [comment]
Still waiting.. 😉

**Comment by lordjancso** [comment]
Is there any news about this?

**Comment by cmmata** [comment]
@lordjancso @dmkuznetsov @nullziu If you want a workaround while this request is not accepted, I have been using @Ener-Getick's fork since June 22 without any problem.

File: Command/RunCommand.php (hunk 2 above)
Comment (nclavaud) [inline Command/RunCommand.php]: @Ener-Getick @schmittjoh what do you think of https://github.com/Ener-Getick/JMSJobQueueBundle/pull/3 ?

Comment (Ener-Getick) [inline Command/RunCommand.php]: @nclavaud i don't think it will work when using `phpunit` :/

**Comment by Ener-Getick** [comment]
It looks like @schmittjoh added sf 3 support [in the latest commits](https://github.com/schmittjoh/JMSJobQueueBundle/commits/master), so let's close this pr :)

**Pull request closed without merging**